src/swarm/neo/connection/RequestOnConnBase.d

```diff
@@ -1014,22 +1014,13 @@ abstract class RequestOnConnBase
                           "supported because it has indirections");
             T value;
 
-            int resume_code = this.receiveAndHandleEvents(
-                (in void[] payload)
-                {
-                    this.outer.connection.message_parser.parseBody!(T)(
-                        payload, value
-                    );
-                }
-            );
-
-            if (resume_code >= 0)
-            {
-                assert(on_other_resume_code !is null,
-                       "User unexpectedy resumed the fiber");
-                on_other_resume_code(resume_code);
-            }
+            NextEventFlags flags;
+            flags |= NextEventFlags.Receive;
+            auto event = this.nextEvent(flags);
+            assert(event.active == event.active.received);
```

Comment:
- Why not `auto event = this.nextEvent(NextEventFlags.Receive);`?
- The assertion is done already when evaluating `event.received`.

Reply:
>Why not auto event = this.nextEvent(NextEventFlags.Receive);?

Right :man_facepalming: 

>The assertion is done already when evaluating event.received.

Ah, indeed.

Reply:
Updated.